src/three-components/TextureUtils.js

```diff
@@ -70,91 +72,110 @@ export default class TextureManager extends EventDispatcher {
+    return generator.renderTarget;
   }
 
   /**
    * @param {string} url
    * @return {Promise<THREE.Texture>}
    */
   async load(url) {
+    const isHDR = HDR_FILE_RE.test(url);
+    const loader = isHDR ? hdrLoader : ldrLoader;
     const texture = await new Promise(
         (resolve, reject) => loader.load(url, resolve, undefined, reject));
     texture.userData = {
       ...userData,
       ...({
         url: url,
-        type: 'Equirectangular',
+        mapping: 'Equirectangular',
       })
     };
 
-    texture.encoding = GammaEncoding;
+    if (isHDR) {
+      texture.encoding = RGBEEncoding;
+      texture.minFilter = NearestFilter;
+      texture.magFilter = NearestFilter;
+      texture.flipY = true;
+    } else {
+      texture.encoding = GammaEncoding;
+    }
+
     return texture;
   }
 
   /**
    * @param {?number} size
    * @return {THREE.Texture}
    */
-  generateDefaultEnvMap(size) {
+  generateDefaultEnvironmentMap(size) {
     const mapSize = size || this.config.synthesizedEnvmapSize;
-    const texture = this.envMapGenerator.generate(mapSize);
-    texture.userData = {
+    const cubemap = this.environmentMapGenerator.generate(mapSize);
+
+    cubemap.userData = {
       ...userData,
       ...({
-        type: 'EnvironmentMap',
+        url: null,
+        mapping: 'Equirectangular',
       })
     };
 
-    return texture;
+    return cubemap;
   }
 
   /**
-   * Returns a { equirect, cubemap } object with the textures
+   * Returns a { skybox, environmentMap } object with the targets/textures
    * accordingly, or null if cannot generate a texture from
-   * the URL.
+   * the URL. `skybox` is a WebGLRenderCubeTarget, and `environmentMap`
+   * is a Texture from a WebGLRenderCubeTarget.
    *
    * @see equirectangularToCubemap with regard to the THREE types.
    * @param {string} url
    * @return {Promise<Object|null>}
    */
-  async toCubemapAndEquirect(url) {
-    let equirect, cubemap;
+  async generateEnvironmentTextures(url) {
+    let equirect, skybox, environmentMap;
     try {
       equirect = await this.load(url);
-      cubemap = await this.equirectangularToCubemap(equirect);
-      return {equirect, cubemap};
-    } catch (e) {
+      skybox = await this.equirectangularToCubemap(equirect);
+
+      // Use the same texture (directly instead of WebGLRenderTargetCube)
+      // for the environment map until #215
+      environmentMap = skybox.texture;
+
+      equirect.dispose();
+
+      return {environmentMap, skybox};
+    } catch (error) {
       if (equirect) {
         equirect.dispose();
       }
-      if (cubemap) {
-        cubemap.dispose();
+      if (skybox) {
+        skybox.dispose();
       }
+      if (environmentMap) {
+        environmentMap.dispose();
+      }
+
       return null;
     }
   }
 
   dispose() {
-    this.envMapGenerator.camera.renderTarget.dispose();
-    this.envMapGenerator = null;
+    this.environmentMapGenerator.camera.renderTarget.dispose();
```

Comment:
Should `EnvironmentMapGenerator` have its own `dispose` method so that we don't have to dig this deeply into its implementation details here?

Reply:
yeah, probably, this is a bit much